Spigot-API-Patches/0232-Item-no-age-no-player-pickup.patch

```diff
@@ -0,0 +1,45 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Alfie Smith <alfie@alfiesmith.net>
+Date: Sat, 7 Nov 2020 01:20:27 +0000
+Subject: [PATCH] Item no age & no player pickup
+
+
+diff --git a/src/main/java/org/bukkit/entity/Item.java b/src/main/java/org/bukkit/entity/Item.java
+index c404a5b8efea7c780db5ddae19456753808abb3d..5acbe1504fcdb93540acde0dcb00a9a6362dfc1e 100644
+--- a/src/main/java/org/bukkit/entity/Item.java
++++ b/src/main/java/org/bukkit/entity/Item.java
+@@ -90,5 +90,34 @@ public interface Item extends Entity {
+      * @param canMobPickup True to allow non-player entity pickup
+      */
+     public void setCanMobPickup(boolean canMobPickup);
++
++    /**
++     * Gets whether the player can pickup the item or not
++     *
++     * @return whether a player can pickup the item
++     */
++    public boolean canPlayerPickup();
++
++    /**
++     * Sets whether the item can be picked up or not. Modifies the pickup delay value to do so.
++     *
++     * @param noPickup Whether the item should be able to be picked up
++     */
++    public void setCanPlayerPickup(boolean noPickup);
```

Comment:
why is the var name negated here? this is rather confusing, if I want `canPlayerPickup` to return false, do I have to set noPickup to true? (would be inverted logic) 

looking at the impl, there seems to be smth wrong, "return item.pickupDelay == noPickupTime;" would return true if a player can't pickup the item? 

idk, maybe its just early morning mini logic, but something seems off, maybe others can take a look

Reply:
I agree with you on the naming - I will change that to canPlayerPickup. Will probably also change the ageing field as @PaulBGD suggested. You're also correct about the implementation, it is also backwards. Will fix later and update the PR.

Reply:
@MiniDigger @Proximyst should be fixed now, though the patch number for server file may be needed to be bumped by one 👍